snuba/querylog/__init__.py

```diff
@@ -38,20 +41,28 @@ def record_query(
             mark_tags={"final": final, "referrer": referrer},
         )
 
-        _add_tags(timer, request)
+        _add_tags(timer, request, extra_data.get("experiments"))
 
 
-def _add_tags(timer: Timer, request: Optional[Request] = None) -> None:
+def _add_tags(
+    timer: Timer,
+    request: Optional[Request] = None,
+    experiments: Optional[Mapping[str, Any]] = None,
+) -> None:
     if Hub.current.scope.span:
         duration_group = timer.get_duration_group()
         sentry_sdk.set_tag("duration_group", duration_group)
         if duration_group == ">30s":
             sentry_sdk.set_tag("timeout", "too_long")
-
         if request is not None:
-            experiments: MutableMapping[str, Any] = request.query.get_experiments()
+            query_experiments: MutableMapping[
+                str, Any
+            ] = request.query.get_experiments()
+            for name, value in query_experiments.items():
+                sentry_sdk.set_tag(name, str(value))
+        if experiments is not None:
             for name, value in experiments.items():
-                sentry_sdk.set_tag(f"exp-{name}", str(value))
+                sentry_sdk.set_tag(name, str(value))
```

Comment:
How come the query experiments are added to tags twice? Don't the represent the same thing?

Reply:
They represent the same thing but will in practice have different values. One is coming from the logical query (the request) and one is coming from the clickhouse query. Those can (and probably will) have different experiments on them.

Reply:
When the logical query is transformed into a ClickHouse query don't the experiments defined on the logical query at that point just get passed along though? (or could they if they don't already) The ClickHouse query could still add more experiments to that original list during the later stages of processing if required.

Reply:
It could, is that to save have two dicts and combining them? The real problem is how to get the experiments off the clickhouse query and into this function, since it originally only had access to the request query.

Reply:
I kind of just assumed it worked that way since the rest of the properties of the query are transformed keeping their original values.

What kind of experiments will we be having here and where do you envisage the experiments being assigned though? It seems to me that we could just decide them all in one place at the start of the pipeline (on the logical query), then that list passed to the ClickHouse query for execution where those already pre-selected experiments may be applied via the relevant query processing or transformation steps.

Reply:
When we start to do some more sophisticated splitting and recombining of query results will having different experiments per Clickhouse query cause any issues?

Reply:
> What kind of experiments will we be having here and where do you envisage the experiments being assigned though?

So the reason I have this change is because the empty tag experiment that is currently running is not recording results since it operates on the Clickhouse query. I don't think we can decide all the experiments on the logical query, because while there are experiments that deal with the logical data (for exampling sampling based on referrer), there are other experiments based on the physical data model (leveraging a bloom filter that exists on a particular promoted column). I don't think we can safely assume which experiments will run on the physical model at the time we are dealing with the logical model.

> When we start to do some more sophisticated splitting and recombining of query results will having different experiments per Clickhouse query cause any issues?

Yes almost certainly. This experiment tracking is in its infancy and needs to be improved as we try more and more complicated experiments. For example I think we will want experiments stored in the querylog so that if a single transaction in Snuba spawns multiple queries, we can see which experiments ran on which queries. Or we could split out each query in Snuba into a separate transaction, so we can tag each query with its experiments and compare them in Discover. Or both!

Reply:
Why don't you just propagate the experiments from the logical to the clickhouse query and just rely on the extra data field which at that point will contain both logical and clikhouse experiments? 
This way this code will look like less confusing as it will have to do one thing instead of two.